instrumentation/spring/spring-rmi-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/rmi/v4_0/server/ServerInstrumentation.java

```diff
@@ -41,49 +43,60 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class InvokeMethodAdvice {
 
-    @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void onEnter(
-        @Advice.This RmiBasedExporter thisObject,
-        @Advice.Argument(0) RemoteInvocation remoteInv,
-        @Advice.Local("otelCallDepth") CallDepth callDepth,
-        @Advice.Local("otelRequest") ClassAndMethod request,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
+    public static class AdviceScope {
+      private final CallDepth callDepth;
+      private final ClassAndMethod request;
+      private final Context context;
+      private final Scope scope;
```

Comment:
maybe it would be better not to use `final` so that you wouldn't need to assign `null` below, wdyt?

Reply:
or maybe use a static method instead of the constructor like down below in `AnnotatedMethodAdvice`